pkg/ipcache/interface.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ipcache
+
+import (
+	"context"
+	"net"
+	"net/netip"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/ipcache/types"
+	store2 "github.com/cilium/cilium/pkg/kvstore/store"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/source"
+	types2 "github.com/cilium/cilium/pkg/types"
+)
+
+type Interface interface {
+	Shutdown() error
+	Lock()
+	Unlock()
+	RLock()
+	RUnlock()
```

Comment:
I think the lock functions should probably also be delimited by a comment that prompts developers to avoid these locks if at all possible. In general locks shoudn't be part of an interface, this is a sign that the abstractions are not fully formed.